### ISSUE TITLE
tailwindcss --watch does not purge unused classes when saving a source file

### ISSUE THREAD

**Issue by sylann** [tailwindlabs/tailwindcss#14290]
**What version of Tailwind CSS are you using?**

v3.4.6

**What build tool (or framework if it abstracts the build tool) are you using?**

None

**What version of Node.js are you using?**

v20.15.1

**What browser are you using?**

Not relevant

**What operating system are you using?**

macOS, Sonoma (14.6.1), Intel X86

**Reproduction URL**

https://github.com/sylann/tailwind-issue-14290

**Describe your issue**

When running tailwindcss with the --watch flag:
```sh
node_modules/.bin/tailwindcss -i ./input.css -o ./output.css --watch
```

1. The first generation in ./output.css is correct
  - ✅ used classes are added
  - ✅ unused classes are removed
2. Subsequent generations (on updates) are incorrect
  - ✅ used classes are added
  - ❌ unused classes are not removed

**Comment by adamwathan** [comment]
Hey! See previous issue: https://github.com/tailwindlabs/tailwindcss/issues/4098

**Comment by sylann** [comment]
Hi @adamwathan,

Sorry for the "kind of" duplicate issue.

After reading your linked issue, and reading the docs again,
although I understand the emphasis you put on performance, I feel like the behavior of --watch is not exactly obvious.

I am tracking the output css file, and the fact that the generated css is not purging unused css classes tricked me for some time before I noticed.

I just recently started using the more recent versions of tailwind.
I see how I could circumvent any problem regarding builds and deployment, but this seems like a bunch of additional work that did not exist in prior usage of tailwindcss.

Am I missing something?
The get started page does not really mention anything and I can't seem to find a recent dedicated page/article. But it's possible I am bad at searching.
